test/pylib/manager_client.py

```diff
@@ -58,6 +58,21 @@ def _driver_update(self) -> None:
         if self.ccluster is not None:
             self.ccluster.control_connection.refresh_node_list_and_token_map()
 
+    async def before_test(self, test_name: str) -> None:
+        """Before a test starts check if cluster needs cycling and update driver connection"""
+        dirty = await self.is_dirty()
+        if dirty:
+            self.driver_close()  # Close driver connection to old cluster
+        await self._request(f"http://localhost/cluster/before-test/{test_name}")
```

Comment:
I think I asked this before but can't find my question (or answer) - what is this "localhost" about? Why is this localhost and not a very specific IP address, which we need to allow multiple instances of this test to run in parallel?

Reply:
> Why do you use "localhost" when communicating with "the harness"?
> 
> I think we should always think about the possibility that the same test might be run in parallel, and allow everything in it, including the harness, to use a unique IP address or a unique port. 

On second thought, I just remembered that you claimed that you were using a UNIX-domain socket - i.e., a unique temporary file - and not TCP. So what is that "localhost" URLs I saw all over the code? How do you get those "localhost" URLs not to communicate with 127.0.0.1 and instead communicate with a UNIX-domain socket?

Reply:
@nyh I haven't tested, but I guess pretty much any domain name would work. What would you like it to be?

Reply:
https://github.com/scylladb/scylladb/pull/10815/files/1bfb80b8bb534238f7f40399c4657b1a60a4663f..8a0345250f901a8830a4a400b73272d34a5afdcb#diff-ac4b42f4b43b922bee79039882129028d018615824be306f1845bfdb86b350d2R40

Reply:
self.conn = aiohttp.UnixConnector(path=self.sock_path)

Reply:
Oh, I understand now. So the hostname is just ignored (and probably gets sent as a "Host:" header if it uses the real HTTP protocol). Maybe it would have been clearer if we had a silly-looking string like "unixsocket" instead of something which looks like it might be real like "localhost". But you're right, I'm happy that localhost is **not** used - what the request looks like is of secondary importance.